# Amazon Dataset Analysis and visualization by Group 4

In [128]:
import pandas as pd
import random
import numpy as np
import re

In [129]:
import pandas as pd
import os
os.chdir("/Users/raghavasaikowlasker/PycharmProjects/AmazonDM")
print("Current Working Directory " , os.getcwd())

df = pd.read_csv("/Users/raghavasaikowlasker/PycharmProjects/AmazonDM/Amazon_Dataset.csv")

# Displays top 5 rows
df.head(5)

# Displays whole table
df

Current Working Directory  /Users/raghavasaikowlasker/PycharmProjects/AmazonDM


uniq_id   manufacturer   price  \
0     eac7efa5dbd3d667f26eb3d3ab504464         Hornby    3.42   
1     348f344247b0c1a935b1223072ef9d8a            ccf    9.99   
2     e33a9adeed5f36840ccc227db4682a36         Hornby   32.19   
3     87bbb472ef9d90dcef140a551665c929         Hornby   69.93   
4     03a35de3f7af9814978e6de645cb8ffa         Hornby  119.50   
...                                ...            ...     ...   
3368  5962c0a2623129d656aac1ee19239dcb         Marvel   11.95   
3369  0c99ecefabf3ec25d4fae53d0646fb3b    Thundercats   11.44   
3370  57c638712b4ecb8dac1ec3004039f1f1  Power Rangers    9.75   
3371  44d6967f083825a5de36ad4865a65bcd         Mattel   22.95   
3372  94d18e86e504bee1e392df7e5738b18a         Bandai   21.20   

      number_of_reviews product_name_new              category  \
0                    15  Hornby 2014 Cat              Hobbies    
1                    17  CLASSIC TOY TRA              Hobbies    
2                     3  Hornby 00 Gauge              Hobbies    
3                    36  Hornby Santa's               Hobbies    
4                     3  Hornby 00 Gauge              Hobbies    
...                 ...              ...                   ...   
3368                 12  Playskool Heroe              Hobbies    
3369                  3  Thundercats 10c              Hobbies    
3370                  7  Power Rangers D              Hobbies    
3371                  3  Batman 1966 TV               Hobbies    
3372                 11  Star Wars 1/72   Characters & Brands    

                              subcategory  Rating  
0            Model Trains & Railway Sets      4.9  
1            Model Trains & Railway Sets      3.9  
2            Model Trains & Railway Sets      4.7  
3            Model Trains & Railway Sets      4.3  
4            Model Trains & Railway Sets      5.0  
...                                   ...     ...  
3368   Collectible Figures & Memorabilia      4.8  
3369   Collectible Figures & Memorabilia      4.7  
3370   Collectible Figures & Memorabilia      4.1  
3371   Collectible Figures & Memorabilia      5.0  
3372                           Star Wars      4.5  

[3373 rows x 8 columns]

In [138]:
import plotly.graph_objects as go
import pandas as pd
e = pd.read_csv("Amazon_Dataset.csv")
df=pd.DataFrame(e)

fig = go.Figure([go.Histogram(x=df['category'], y=df['manufacturer'].unique())])
fig.update_layout(title_text="Number of manufacturers for each category  ")
fig.update_xaxes(title_text="Category")
fig.update_yaxes(title_text="Number of manufacturer")
fig.show()

In [131]:
import plotly.express as px
import pandas as pd
import plotly.graph_objects as go

e = pd.read_csv ('/Users/raghavasaikowlasker/PycharmProjects/AmazonDM/Amazon_Dataset.csv')
df=pd.DataFrame(e)
filter_reviews = df[df['number_of_reviews'] > 300]
fig = px.pie(filter_reviews, values='number_of_reviews', names='manufacturer',title='Total Reviews for Each Manufacturer',color_discrete_sequence=px.colors.sequential.RdBu
             , hover_data=['category'], labels={'reviews':'number_of_reviews'})
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [132]:
import pandas as pd

df = pd.read_csv("Amazon_Dataset.csv")


# Displays top 5 rows
df.head(5)

print('Number of Unique elements in column manufacturers and category')
uniqueValue = (df['manufacturer'].nunique(),df['category'].nunique(),df['product_name_new'].nunique(),df['subcategory'].nunique())
print(uniqueValue)
unum = (df.manufacturer.unique(),df.category.unique())
print('Unique elements in column manufacturers and category')
unum


Number of Unique elements in column manufacturers and category
(1141, 32, 2699, 125)
Unique elements in column manufacturers and category


(array(['Hornby', 'ccf', 'Gaugemaster', ..., 'SMART SHOPPING', 'Abystyle',
        'Square-Enix'], dtype=object),
 array(['Hobbies ', 'Characters & Brands ', 'Bags ', 'Games ',
        'Arts & Crafts ', 'Figures & Playsets ', 'Home Accessories ',
        'Die-Cast & Toy Vehicles ', 'Storage, Cleaning & Ring Sizers ',
        'Bedding & Linens ', 'Sports Toys & Outdoor ', 'Camping & Hiking ',
        'Pretend Play ', 'Fancy Dress ', 'Women ', 'Electronic Toys ',
        'Baby & Toddler Toys ', 'Party Supplies ', 'Dolls & Accessories ',
        'Educational Toys ', 'Sweets, Chocolate & Gum ',
        'Musical Toy Instruments ', 'Men ', 'Cooking & Dining ',
        'Medication & Remedies ', 'Puppets & Puppet Theatres ',
        'Indoor Lighting ', 'Jigsaws & Puzzles ', 'Office Supplies ',
        'Pens, Pencils & Writing Supplies ', 'Worlds Apart ', 'Gardening '],
       dtype=object))

# 32 categories having 125 subcategories with over 1141 manufacturers selling over 2669 products.

In [133]:
import plotly.graph_objects as go
import pandas as pd
import plotly.express as px
e = pd.read_csv("Amazon_Dataset.csv")
df=pd.DataFrame(e)
fig = px.histogram(df, y= df['Rating'], x="category", color_discrete_sequence=['indianred'])
fig.update_layout(title_text="Highest gross Ratings for Each Category")
fig.update_xaxes(title_text="Category")
fig.update_yaxes(title_text="Total Ratings")
fig.show()

In [134]:
import plotly.graph_objects as go
import pandas as pd
import plotly.express as px
e = pd.read_csv("Amazon_Dataset.csv")
df=pd.DataFrame(e)

fig = px.pie(df, values='Rating', names='category',title='Gross Ratings for each Category'
             , hover_data=['category'], labels={'reviews':'number_of_reviews'})
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()



In [135]:
import plotly.express as px
import math
import dash
e = pd.read_csv("Amazon_Dataset.csv")
df=pd.DataFrame(e)
fig = px.treemap(df, path=['category','subcategory'], values='price',
                  color='price', title = 'Average Pricing with respect Category/Subcategory')
fig.show()

/Users/raghavasaikowlasker/Documents/GitHub/metric-models/venviron/lib/python3.8/site-packages/plotly/express/_core.py:1637: FutureWarning:

The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.

/Users/raghavasaikowlasker/Documents/GitHub/metric-models/venviron/lib/python3.8/site-packages/plotly/express/_core.py:1637: FutureWarning:

The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.

